src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryFormatterEventSource.cs

```diff
@@ -0,0 +1,158 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.Threading;
+
+namespace System.Runtime.Serialization.Formatters.Binary
+{
+    [EventSource(
+        Name = "System.Runtime.Serialization.Formatters.Binary.BinaryFormatterEventSource",
+        LocalizationResources = "FxResources.System.Runtime.Serialization.Formatters.SR")]
+    internal sealed class BinaryFormatterEventSource : EventSource
+    {
+        private const int EventId_SerializationStarted = 10;
+        private const int EventId_SerializationEnded = 11;
+        private const int EventId_SerializingObject = 12;
+        private const int EventId_DeserializationStarted = 20;
+        private const int EventId_DeserializationEnded = 21;
+        private const int EventId_DeserializingObject = 22;
+
+        // Used to keep track of whether a write operation is in progress. It's
+        // possible the listener itself uses BinaryFormatter to write to a log,
+        // and if this is the case we suppress our own logging events so that we
+        // enter an infinite recursion scenario.
+        private static readonly AsyncLocal<bool> _writeInProgress = new AsyncLocal<bool>();
+
+        public static readonly BinaryFormatterEventSource Log = new BinaryFormatterEventSource();
+
+        private BinaryFormatterEventSource()
+        {
+        }
+
+        [Event(EventId_SerializationStarted, Opcode = EventOpcode.Start, Keywords = Keywords.Serialization, Level = EventLevel.Informational)]
```

Comment:
Are these start/stop events ever going to nest recursively intentionally? If so I recommend adding ActivityOptions = EventActivityOptions.Recursive so that anyone who decides to enable activity tracking will be able to track the nesting of the calls.